### ISSUE TITLE
[RetinaNet] I need more explanation in "class RetinaNet(keras.Model)"

### ISSUE THREAD

**Issue by yhu0409** [keras-team/keras-io#444]
Hello, now I'm customizing the example RetinaNet in here. But I don't understand this part.

As I know, this parameter "N" in this part means image's height. For example, if input image's shape is (256, 256, 3), this N should be 256. Furthermore, the shape of output through "box_head" about "p7_output" is (36, 2, 2).

But the error occurs in "tf.reshape(self.box_head(feature), [N, -1, 4])" because 36 * 2 * 2 = 144 is less than "N" which is 255.

I know I misunderstood. But I don't know where I misunderstood.

so please help me.

#    def call(self, image, training=False):
        features = self.fpn(image, training=training)
        N = tf.shape(image)[0]
        cls_outputs = []
        box_outputs = []
        for feature in features:
            box_outputs.append(tf.reshape(self.box_head(feature), [N, -1, 4]))
            cls_outputs.append(
                tf.reshape(self.cls_head(feature), [N, -1, self.num_classes])
            )
        cls_outputs = tf.concat(cls_outputs, axis=1)
        box_outputs = tf.concat(box_outputs, axis=1)
        return tf.concat([box_outputs, cls_outputs], axis=-1)

**Comment by yhu0409** [comment]
I understood this N means batch_size.
